libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m

```diff
@@ -387,14 +368,139 @@ - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command {
     } command:command];
 }
 
+- (void)pgGetAllGlobalStores:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        NSArray *allStoreNames = [SFSmartStore allGlobalStoreNames];
+        NSMutableArray *result = [NSMutableArray array];
+        if (allStoreNames.count >0 ) {
+            for(NSString *storeName in allStoreNames) {
+                NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+                [dictionary setValue:storeName forKey:kStoreName];
+                [dictionary setValue:@YES forKey:kIsGlobalStoreArg];
+                [result addObject:dictionary];
+            }
+        }
+        return [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:result];
+     }
+     command:command];
+}
+
+- (void)pgGetAllStores:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        NSArray *allStoreNames = [SFSmartStore allStoreNames];
+        NSMutableArray *result = [NSMutableArray array];
+        if (allStoreNames.count >0 ) {
+            for(NSString *storeName in allStoreNames) {
+                NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+                [dictionary setValue:storeName forKey:kStoreName];
+                [dictionary setValue:@YES forKey:kIsGlobalStoreArg];
+                [result addObject:dictionary];
+            }
+           
+        }
+        return [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:result];
+    } command:command];
+
+}
+
+- (void)pgRemoveStore:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        BOOL isGlobal = [self isGlobal:argsDict];
```

Comment:
The underlying smartstore API(s) define different static methods to invoke based on whether the store is global or not. There could be value to adding a SFStoreConfig class only if we were to  add more generic add/remove store apis to the SFSmartStore that are based on this StoreConfig object used as input. The StoreConfig class would at the least have these  2 properties (storeName,isGlobalStore).